src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -1121,7 +1125,7 @@ impl EventProcessor {
 
             let event = Event::WindowEvent {
                 window_id,
-                event: WindowEvent::CursorMoved { device_id, position },
+                event: WindowEvent::CursorMoved { device_id, position, r#type: CursorType::Mouse },
```

Comment:
Okay I figured this one out as well.
After digging around in Chromium for a while I found [how they determined if this was a stylus or not](https://source.chromium.org/chromium/chromium/src/+/c85f08af47c25bc5eb4a7f81c0ac0620d8d767ca:ui/events/devices/x11/device_data_manager_x11.cc;l=198-200).

After some digging, I also found that [Glazier was doing something quite similar](https://github.com/linebender/glazier/blob/8d253c3d4ad12c1dd491895bf3eb8ef1728d01b6/src/backend/x11/pointer.rs#L146-L157), in addition to detecting erasers (in a hacky way, but it works). There is a lot to learn from them when actually implementing the backends for Pens/Stylus's.

Additionally, after some more digging into X11, I found that there is an alternative to figuring out the type: [`XListInputDevices`](https://www.x.org/releases/current/doc/man/man3/XListInputDevices.3.xhtml). That's right, using the old XInput version and now way to query individual devices. This exposes what they call "Device Types", which unfortunately don't include pens, but they let you recognize the drawing tablet,  which works as well.

I found that [Chromium was doing something similar for touchpads, but not for pens](https://source.chromium.org/chromium/chromium/src/+/0a212cf3cbde2f8639dc3238dfe4762969bd170d:ui/events/devices/x11/device_data_manager_x11.cc;l=251-258) ([and a comment in a much older revision explaining this as well](https://source.chromium.org/chromium/chromium/src/+/9054218f820a9d0502c24db35837c81a8f4ffa73:ui/events/devices/x11/touch_factory_x11.cc;l=86-100)). So maybe this will be useful for us in the future.